src/app/meme-dialog/meme-dialog.component.ts

```diff
@@ -0,0 +1,70 @@
+import { Component, Inject, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { MatDialogModule, MAT_DIALOG_DATA, MatSnackBar } from '@angular/material';
+import { Utils } from '../utils';
+import { MemeService } from '../api/meme.service';
+
+@Component({
+  selector: 'app-meme-dialog',
+  templateUrl: './meme-dialog.component.html',
+  providers: [MatDialogModule],
+  styleUrls: ['./meme-dialog.component.scss']
+})
+export class MemeDialogComponent implements OnInit {
+  username: string;
+  imageLink: string;
+  totalVote: number;
+  myVote: number;
+  memeId: number;
+
+  @Output() notifyCard: EventEmitter<number> = new EventEmitter<number>();
+
+  constructor(private memeService: MemeService,
+              private snackBar: MatSnackBar,
+              @Inject(MAT_DIALOG_DATA) public data: any) {
```

Comment:
You can just pass `memeId` from the parent into `data` and then use the `memeService` to get the meme details.

You probably implemented this for performance reasons since it doesn't make an extra request, but https://github.com/MemePlace/Frontend/pull/73 mitigates that issue by caching `getMemeDetails` so the result should be instant.

Reply:
Done